### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by MohamedHNoor**
### Classes Created:

- [x] Nameable 
- [x] Decorator
- [x] CapitalizeDecorator
- [x] TrimmerDecorator

### CODE REVIEW

File: capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+require_relative './person'
+
+class CapitalizeDecorator < Decorator
+  def initialize(nameable)
+    super
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name.upcase
```
Comment (TedLivist) [inline capitalize_decorator.rb]: - [ ] We want to capitalize the name, not make it all upper cased

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @MohamedHNoor,

Good job so far!
✔️ Implementation of inheritance
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: capitalize_decorator.rb (hunk 1 above)
Comment (MohamedHNoor) [reply capitalize_decorator.rb]: done

**Review by julie-ify — APPROVED** [review]
Hi @MohamedHNoor, 

## Highlights!
- Capitalizing the name ✔️
- Good code organization and working app ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**